### ISSUE TITLE
[java] ArrayIsStoredDirectly false positive with private constructor/methods

### ISSUE THREAD

**Issue by kevemueller** [pmd/pmd#2622]
**Affects PMD Version:** 
6.21.0.
**Rule:**
[ArrayIsStoredDirectly](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#arrayisstoreddirectly)

**Description:**
The rule should not trigger if the constructor/method under analysis is private.
Private constructors/methods do not get "user supplied" arrays.

**Code Sample demonstrating the issue:**

```java
public class AISD {
    private final byte[] buf;
    private AISD(final byte[] buf) {
        this.buf = buf;  // this is not a violation!!
    }
    public AISD of(final byte[] buf) {
        return new AISD(buf.clone());
    }
}
```

**Expected outcome:**
PMD reports a violation, where there is none.
PMD should check the visibility of the constructor and skip the check.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Maven

**Comment by kevemueller** [comment]
Cool. I see you extended my example with a setter. Please consider dropping the final from the field, so the example would actually compile.
